pennylane/workflow/construct_batch.py

```diff
@@ -355,14 +457,114 @@ def batch_constructor(*args, **kwargs) -> tuple[QuantumScriptBatch, Postprocessi
         params = initial_tape.get_parameters(trainable_only=False)
         initial_tape.trainable_params = qml.math.get_trainable_indices(params)
 
-        config = qml.workflow.construct_execution_config(qnode, resolve=False)(*args, **kwargs)
-        # pylint: disable = protected-access
-        config = qml.workflow.resolution._resolve_execution_config(
-            config, qnode.device, tapes=(initial_tape,)
+        level_slice_initial = _interpret_level_initial(
+            level, num_user_transforms
+        )  # This should be fine, since the case where `has_gradient_expand==True` only increase 1 to the end of level slice
+        program = user_program[level_slice_initial]
+        if _is_within_user_bounds(level, num_user_transforms):
+            # If the level slice is fully contained within user transforms, we can return early
+            return program((initial_tape,))
+
+        user_transformed_tapes, user_post_processing = program((initial_tape,))
+        #### User transforms finished #####
+        # The new config process we would like to use.
+        mcm_config = qml.devices.MCMConfig(
+            postselect_mode=qnode.execute_kwargs["postselect_mode"],
+            mcm_method=qnode.execute_kwargs["mcm_method"],
         )
-        gradient_fn = config.gradient_method
-        program = get_transform_program(qnode, level=level, gradient_fn=gradient_fn)
+        execution_config = _make_execution_config(
+            qnode, qnode.diff_method, mcm_config
+        )  # pylint: disable = protected-access
+
+        ###### Resolution of the execution config ######
+        execution_config = _resolve_execution_config(
+            execution_config,
+            qnode.device,
+            tapes=user_transformed_tapes,  # Use the user-transformed tapes
+        )
+
+        # Use _setup_transform_program like execute() does
+        outer_transform_program, inner_transform_program = _setup_transform_program(
+            qnode.device,
+            execution_config,
+            cache=qnode.execute_kwargs["cache"],
+            cachesize=qnode.execute_kwargs["cachesize"],
+        )
+        full_transform_program = outer_transform_program + inner_transform_program
+
+        gradient_fn = execution_config.gradient_method
```

Comment:
Do we really need an explanatory variable for this?

Reply:
https://github.com/PennyLaneAI/pennylane/pull/7461/commits/94d04e52d73854016d013467bbba57b871b97196